src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -524,4 +564,64 @@ private static List<DetailAST> getMethodAnnotationsList(DetailAST methodDef) {
         return annotationsList;
     }
 
+    /**
+     * Represents the Java language level.
+     */
+    private enum JavaLanguageLevel {
+
+        /**
+         * Represents Java language level less than 17.
+         */
+        LEVEL_LT_17(16),
+
+        /**
+         * Represents Java language level 17.
+         */
+        LEVEL_17(17),;
+
+        /**
+         * The value of the language level.
+         */
+        private final int value;
+
+        /**
+         * Creates a new instance.
+         *
+         * @param value the value of the language level
+         */
+        JavaLanguageLevel(int value) {
+            this.value = value;
+        }
+
+        /**
+         * Returns the language level based on the JDK version.
+         *
+         * @param jdkVersion the JDK version
+         * @return the java language level
+         */
+        private static JavaLanguageLevel getLevel(String jdkVersion) {
+            final JavaLanguageLevel level;
+            final int jdk17 = 17;
+            if (jdkVersion.startsWith("1.")
+                    || Integer.parseInt(jdkVersion) < jdk17) {
+                level = LEVEL_LT_17;
+            }
+            else {
+                level = LEVEL_17;
+            }
+            return level;
```

Comment:
As I was talking with @nrmancuso , this enumeration just seems weird to me.

`LEVEL_LT_17(16)` is odd, since now we say its 16 for anything less than 17.
Similarly, we say anything above 17 is just 17.

This `getLevel` method also never fails. I could pass in a property value of `abc` and it would pass and assume I am on 17 (or above). The user would never get any feedback on an issue with what they set it as, and in our case, we don't set it ourselves.

I understand we don't need all possible exact values, but it also seems just weird we would do this for what mostly seems like a simple integer. Even if we add validation to this method to ensure the property value is valid, it seems we could just as easily use the code to manage an `int` over an enumeration.

I don't see the code getting more complex with a simple `int`. `isGreaterThanOrEqual` would just turn into `>=`. We would also not need to worry about breaking compatibility here with future updates.

Reply:
done